docs: update quickstart and references for embedded replica updates (#144)
tursodatabase/turso-docs: sdk/rust/quickstart.mdx

  </Step>

  <Step title="Connect">

You must first create a `Database` object and then open a `Connection` to it:

<AccordionGroup>
  <Accordion title="Embedded Replicas">

```rust
use libsql::Builder;

let url = env::var("LIBSQL_URL").expect("LIBSQL_URL must be set");
let token = env::var("LIBSQL_AUTH_TOKEN").unwrap_or_default();

let mut db = Builder::new_remote_replica("local.db", &url, &token).build().await.unwrap();
let conn = db.connect().unwrap();
```

  </Accordion>
  <Accordion title="Local only">

```rust
use libsql::Builder;

let db = Builder::new_local("local.db").build().await?;
let conn = db.connect().unwrap();
```

  </Accordion>
  <Accordion title="Remote only">

```rust
use libsql::Builder;

let url = env::var("LIBSQL_URL").expect("LIBSQL_URL must be set");
let token = env::var("LIBSQL_AUTH_TOKEN").unwrap_or_default();

let db = Builder::new_remote(&url, &token).build().await?;
let conn = db.connect().unwrap();
```

  </Accordion>
</AccordionGroup>

  </Step>

  <Step title="Execute">

You can execute a SQL query against your existing database by calling `execute()`:

```rust

> conn.execute("SELECT * FROM users").await.unwrap();

Reviewer: ```suggestion
conn.execute("SELECT * FROM users", ()).await.unwrap();
```

Reviewer: The `()` unit type is used as a place holder for no params.

```

If you need to use placeholders for values, you can do that:

<CodeGroup>

```rust Positional

> conn.execute("SELECT * FROM users WHERE id = ?1", params![1]).await?;

Reviewer: ```suggestion
conn.execute("SELECT * FROM users WHERE id = ?1", libsql::params![1]).await?;
```

```

```rust Named
conn.execute("INSERT INTO users (name) VALUES (:name)", libsql::named_params! { ":name": "Iku" }).await?;
```

</CodeGroup>

  </Step>

  <Step title="Sync (Embedded Replicas only)">

> When using embedded replicas you should call `sync()` on the connector to sync your local database with the primary database, unless you are using `periodic_sync`:

Reviewer: ```suggestion
When using embedded replicas you should call `sync()` on the database type to sync your local database with the primary database, unless you are using `periodic_sync` (though there is no issue with calling `sync` with `periodic_sync` enabled):
```
